include/nbl/asset/filters/CBlitImageFilter.h

```diff
@@ -152,6 +152,84 @@ class CBlitImageFilter : public CImageFilter<CBlitImageFilter<Swizzle,Dither,Nor
+					// weights when eventually `windowSample` gets multiplied by them later in
+					// `CFloatingPointSeparableImageFilterKernelBase<CRTP>::sample_functor_t<PreFilter,PostFilter>::operator()`
+					// this exists only because `evaluateImpl` expects a pre filtering step.
+					auto dummyLoad = [](value_type* windowSample, const core::vectorSIMDf& unused0, const core::vectorSIMDi32& unused1, const IImageFilterKernel::UserData* unused2) -> void
+					{
+						for (auto h = 0; h < MaxChannels; h++)
+							windowSample[h] = value_type(1);
+					};
+
+					// There's no real reason that this should have `MaxChannels` number of elements except for the fact that
+					// `CFloatingPointSeparableImageFilterKernelBase<CRTP>::sample_functor_t<PreFilter,PostFilter>::operator()` expects that.
+					// So this, for now, just replicates the same value in all 4 channels.
+					value_type kernelWeight[MaxChannels];
+					// actually used to put values in the LUT
+					auto dummyEvaluate = [&kernelWeight](const value_type* windowSample, const core::vectorSIMDf& unused0, const core::vectorSIMDi32& unused1, const IImageFilterKernel::UserData* unused2) -> void
+					{
+						for (auto h = 0; h < MaxChannels; h++)
+							kernelWeight[h] = windowSample[h];
+					};
+
+					auto computeForAxis = [state, &fScale, &phaseCount, &dummyLoad, &dummyEvaluate, &kernelWeight, &axisOffsets](const asset::IImage::E_TYPE axis, const auto& kernel)
+					{
+						if (axis > state->inImage->getCreationParameters().type)
+							return;
+
+						const auto windowSize = kernel.getWindowSize()[axis];
+
+						IImageFilterKernel::ScaleFactorUserData scale(1.f / fScale[axis]);
+						const IImageFilterKernel::ScaleFactorUserData* otherScale = IImageFilterKernel::ScaleFactorUserData::cast(state->kernelX.getUserData());
+
+						if (otherScale)
+						{
+							for (auto k = 0; k < MaxChannels; k++)
+								scale.factor[k] *= otherScale->factor[k];
+						}
+
+						value_type* phaseSupportLUTPixel = reinterpret_cast<value_type*>(state->scratchMemory + getPhaseSupportLUTByteOffset(state) + axisOffsets[axis]);
+						for (uint32_t i = 0u; i < phaseCount[axis]; ++i)
+						{
+							core::vectorSIMDf tmp;
+							tmp[axis] = float(i) + 0.5f;
+
+							core::vectorSIMDi32 windowCoord;
+							windowCoord[axis] = kernel.getWindowMinCoord(tmp * fScale, tmp)[axis];
+
+							float relativePos = tmp[axis] - float(windowCoord[axis]); // relative position of the last pixel in window from current (ith) output pixel having a unique phase sequence of kernel evaluation points
+
+							for (int32_t j = 0; j < windowSize; ++j)
+							{
+								core::vectorSIMDf tmp(relativePos, 0.f, 0.f);
+								kernel.evaluateImpl(dummyLoad, dummyEvaluate, kernelWeight, tmp, windowCoord, &scale); // `windowCoord` is unused
+								phaseSupportLUTPixel[i * windowSize + j] = kernelWeight[0];
+								relativePos -= 1.f;
```

Comment:
ok I see that forward iterated LUT entries store the kernel entries in reverse, you need to document (comment that) so that anyone reading the `execute()` function doesn't go "thIs cOdE Is wrOng It ImpLEmEnts cOrrELAtIOn nOt cOnvOLUtIOn!"

Reply:
https://github.com/Devsh-Graphics-Programming/Nabla/pull/140#discussion_r863620797

Reply:
Resolved in #343.